Components/Components.md

```diff
@@ -1,141 +1,313 @@
-- [Concurrent Mode and Suspense](#concurrent-mode-and-suspense)
+- [Concurrent Rendering and Suspense](#concurrent-rendering-and-suspense)
 - [Memoization with `React.memo` and `useMemo`](#memoization-with-reactmemo-and-usememo)
 - [Fragments](#fragments)
 - [Portals](#portals)
 
-In React, components serve as the **building blocks** of any application. They allow you to create reusable and modular pieces of UI. Let's dive into the world of React components:
-
+---
 
 ## What Are Components?
 
-- **Components** are independent and reusable bits of code.
-- They represent different parts of a web page and contain both structure and behavior.
-- Think of components as Lego blocks that you can assemble to create your entire UI.
+- **Components** are independent, reusable pieces of code that represent parts of a UI.
+- They can manage their own state and behavior or receive data from their parent components.
+- Components help build modular, maintainable, and scalable applications.
+- With React 18, components now benefit from **automatic batching**, where state updates inside promises or event handlers are batched to avoid unnecessary re-renders.
+
+---
 
 ## Types of Components
 
-1. **Functional Components**:
-   - These are like JavaScript functions that accept **props** (properties) and return a React element.
-   - Simple, concise, and easy to write.
-   - Example:
-     ```jsx
-     function Welcome(props) {
-       return <h1>Hello, {props.name}!</h1>;
-     }
-     ```
-
-2. **Class Components**:
-   - More complex than functional components.
-   - Can show inheritance and access data of other components.
-   - Must include `extends React.Component`.
-   - May include lifecycle methods such as componentDidMount, componentDidUpdate, and componentWillUnmount for managing component lifecycle and side effects.
-   - Example:
-     ```jsx
-     import React from 'react';
-     class Welcome extends React.Component {
-       render() {
-         return <h1>Hello, {this.props.name}!</h1>;
-       }
-     }
-     ```
-
-`Note` - Function based components are commonly used and we are going to focus mostly on development using fucntion based components
-
-## Nesting Components
-
-- You can nest components inside one another.
-- This helps create more complex UIs and eliminates redundancy.
-- Components nested inside parent components are called **child components**.
-
-Example:
+### 1. Functional Components
+- Modern React development focuses on **functional components**.
+- Functional components are JavaScript functions that accept **props** (properties) and return JSX.
+- They became more powerful with the introduction of **Hooks**, allowing for state management, lifecycle methods, and more.
+
+**Example:**
 ```jsx
-function App() {
-  return (
-    <div>
-      <Header />
-      <MainContent />
-      <Footer />
-    </div>
-  );
+import React from 'react';
+
+function Welcome({ name }) {
+  return <h1>Hello, {name}!</h1>;
 }
```

Comment:
Please explain the props used inside the component for clear and better understanding